### PR TITLE
fix panic when get job's elastic resource

### PR DESCRIPTION
**Pull request by lowang-bh**
Fix: #3105 (scheduler panic when calculate the elastic resource) via first commit
Fix: #3127 (panic when guarantee greater than allocatable resource) via second commit

### CODE REVIEW

File: pkg/scheduler/api/job_info.go [hunk 1]
```diff
@@ -572,12 +572,15 @@ func (ji *JobInfo) DeductSchGatedResources(res *Resource) *Resource {
...
+// GetElasticResources returns those partly resources in allocated which are more than its minResource
 func (ji *JobInfo) GetElasticResources() *Resource {
-	minResource := ji.GetMinResources()
-	if ji.Allocated.LessEqualPartly(minResource, Zero) {
+	if ji.Allocated == nil {
 		return EmptyResource()
 	}
-	return ji.Allocated.Clone().Sub(minResource)
+	minResource := ji.GetMinResources()
+	elastic := ExceedPart(ji.Allocated, minResource)
```
Comment (Monokaix) [inline pkg/scheduler/api/job_info.go:581]: exceed -> exceeded is better?

**Comment by hwdef** [comment]
Does the capacity plugin need to be handled?

**Comment by lowang-bh** [comment]
> Does the capacity plugin need to be handled?

Have added to it.

File: pkg/scheduler/api/job_info.go (hunk 1 above)
Comment (lowang-bh) [reply pkg/scheduler/api/job_info.go]: Done

**Comment by Monokaix** [comment]
cc @hwdef

**Review by hwdef — COMMENTED** [review]
/lgtm

**Pull request merged**